invenio_app_ils/circulation/transitions/transitions.py

```diff
@@ -0,0 +1,42 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2020 CERN.
+#
+# invenio-app-ils is free software; you can redistribute it and/or modify it
+# under the terms of the MIT License; see LICENSE file for more details.
+
+"""Circulation custom transitions."""
+
+import arrow
+from invenio_circulation.transitions.transitions import (
+    ItemOnLoanToItemOnLoan, ToItemOnLoan)
+
+from invenio_app_ils.holidays.api import next_open_after
+
+
+def _update_loan_end_date(loan):
+    """Update the end date to the next opening day."""
+    current_end_date = arrow.get(loan["end_date"]).date()
+    new_end_date = next_open_after(loan["pickup_location_pid"], current_end_date)
```

Comment:
I would always return a DateTime obj here, even if it is the same, to avoid the if:
```suggestion
    loan["end_date"] = get_end_date_library_is_open(loan["pickup_location_pid"], current_end_date)
```

Reply:
The branch is indeed not needed anymore, it used to be in the case a new date would not be found (all weekdays closed).